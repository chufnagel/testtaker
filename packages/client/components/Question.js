import PropTypes from "prop-types";

const Question = ({ id, text }) => <h5 className="question">{`${id}. ${text}`}</h5>;

Question.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string,
};

Question.defaultProps = {
  id: 1,
  text: "Lorem ipsum",
};

export default Question;
