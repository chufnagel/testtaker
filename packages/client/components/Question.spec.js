/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import Question from "./Question";

// Question.defaultProps = {
//   id: 1,
//   text: "Lorem ipsum",
// };

describe("Question", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Question />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("Question matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
