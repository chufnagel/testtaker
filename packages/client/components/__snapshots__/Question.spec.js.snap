// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Question matches snapshot 1`] = `
<h5
  className="question"
>
  1 Lorem ipsum
</h5>
`;
